timon/db/backends.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+# #############################################################################
+# Copyright : (C) 2023 by MHComm. All rights reserved
+#
+# __author__ = "Quentin Laymajoux"
+# __email__ = "info@mhcomm.fr"
+#
+# Name       : timon.db.backends
+"""
+Summary: Backend Classes to use in dbstore
+"""
+# #############################################################################
+import logging
+from datetime import datetime
+from queue import Queue
+from threading import Event
+from threading import Lock
+
+from peewee import SqliteDatabase
+
+logger = logging.getLogger(__name__)
+
+
+class BaseBackend():
+    """
+    This class is unuseful, but helps to know which funcs must be implemented
+    in backends
+    """
+
+    def __init__(self, **db_cfg):
+        raise NotImplementedError("BaseBackend must be inherited")
+
+    def stop(self):
+        """
+        Cleanly stop the backend
+        """
+        raise NotImplementedError("Backend stop func must be implemented")
+
+    def start(self, probenames):
+        """
+        Setup and start the backend
+        """
+        raise NotImplementedError("Backend setup func must be implemented")
+
+    def get_probe_results(self, probename):
+        """
+        Get probe results for a given probename
+        Rslts is a list of dict ordered by datetime
+        """
+        raise NotImplementedError(
+            "Backend get_probe_results func must be implemented")
+
+    def store_probe_result(self, probename, timestamp, msg, status):
+        """Store a probe result
+
+        Args:
+            probename (str): name of the probe
+            timestamp (int|float): timestamp if the probe run
+            msg (str): probe rslt message
+            status (str): status of the probe result
+        """
+        raise NotImplementedError(
+            "Backend store_probe_result func must be implemented")
+
+
+class PeeweeBaseBackend(BaseBackend):
+    """
+    Store probe results in a db via peewee ORM
+    CAUTION: must be inherited and _get_db func must be
+    implemented.
+    """
+    def __init__(self, **db_cfg):
+        self.rsltqueue = Queue(maxsize=10000)
+        self.stopevent = Event()
+        self.waitevent = Event()
+        self.queuelock = Lock()
+        self.thread_store = None
```

Comment:
store_thread

Reply:
hmm I stop reviewing.
I think there's some of yesterday's changes that are still not pushed